streams/src/test/java/org/apache/kafka/streams/integration/StreamsUncaughtExceptionHandlerIntegrationTest.java

```diff
@@ -177,6 +178,7 @@ public void shouldShutdownMultipleThreadApplication() throws InterruptedExceptio
         testShutdownApplication(2);
     }
 
+    @Ignore // TODO -- re-enable this test after KAFKA-12537
```

Comment:
It would be best to merge https://github.com/apache/kafka/pull/10387 first then we don't have to undo this ignore.

Reply:
I suspect whichever one is merged first will result in small conflicts on the other since they both touch `StreamThread`. But maybe you can remove the changes in StreamThread from your PR since I think they're encompasses by this one, and I'll remove the `Ignore` so they can both be merged once the build passes -- WDYT?

Reply:
Or, feel free to just pull my changes into your PR and I can close this one